- show references, cleanup

AleksandraKozubal opened the pull request:
Fixed links to case studies, so they now show both seeded and added images. Connected references from database to the home page. Also did some cleanup in files.

EwelinaSkrzypacz reviewed (commented):
1. When no references are published, I see error

![image](https://github.com/blumilksoftware/website/assets/56546832/ef845cc7-5954-4aa4-93e9-bbbc7f91d4d7)

2. When there is two references, something strange is happen

https://github.com/blumilksoftware/website/assets/56546832/9d796bce-d54f-4245-a01f-7475f62e9a20

3. And on mobile we will always show one reference? Also, I see error in console, but I think it is because I don't have any image in storage/factory called reference.jpg

![image](https://github.com/blumilksoftware/website/assets/56546832/858b0fb1-5b12-4910-a04f-5b595e8043b0)

4. Why I can't click "Czytaj więcej"?

5. When I write too long name of reference author, I see error:

![image](https://github.com/blumilksoftware/website/assets/56546832/05deb6bc-dc46-4464-9d75-3b9be653150e)

kamilpiech97 commented on resources/views/home.blade.php:
```diff
@@ -4,7 +4,7 @@
     $caseStudy = App::currentLocale()=='en' ? json_decode(file_get_contents(public_path('main_case_study.json')), true)['en'] : json_decode(file_get_contents(public_path('main_case_study.json')), true)['pl'];
     $articles = json_decode(file_get_contents(public_path('articles.json')), true);
-
+//dd(count($references));
```
:eyes:

AleksandraKozubal replied:
oops, deleted:D

kamilpiech97 commented on resources/views/home.blade.php:
```diff
@@ -253,7 +253,7 @@ class="tech-icon h-full w-12 m-auto grayscale hover:grayscale-0 object-contain t
         </section>
 
-        <section class="px-[10%] md:px-[15%] text-center items-center sm:pb-36 relative">
+        <section class=" @if(count($references)==0) hidden @endif px-[10%] md:px-[15%] text-center items-center sm:pb-36 relative">
```
https://laravel.com/docs/11.x/blade#conditional-classes

AleksandraKozubal replied:
changed

AleksandraKozubal commented:
>     1. When no references are published, I see error
> 
>     2. When there is two references, something strange is happen
>
>     3. And on mobile we will always show one reference? Also, I see error in console, but I think it is because I don't have any image in storage/factory called reference.jpg
> 
>     4. Why I can't click "Czytaj więcej"?
> 
>     5. When I write too long name of reference author, I see error:


1,2. fixed
3. Yes, that was designed this way. Added those files, should be ok now
4. I didn't do it because of current design changes - designer is not sure if it will be still there, so it's better to wait and make it when needed
5. fixed

kamilpiech97 commented on resources/views/home.blade.php:
```diff
@@ -2,11 +2,8 @@
-    $caseStudiesFile = App::currentLocale()=='en' ? 'case_studies_en.json' : 'case_studies_pl.json';
-    $caseStudy = json_decode(file_get_contents(public_path($caseStudiesFile)), true)[0];
-    $testimonials = json_decode(file_get_contents(public_path('testimonials.json')), true);
+    $caseStudy = App::currentLocale()=='en' ? json_decode(file_get_contents(public_path('main_case_study.json')), true)['en'] : json_decode(file_get_contents(public_path('main_case_study.json')), true)['pl'];
```
:eyes:

kamilpiech97 commented on resources/views/home.blade.php:
```diff
@@ -252,7 +250,7 @@ class="tech-icon h-full w-12 m-auto grayscale hover:grayscale-0 object-contain t
         </section>
 
-        <section class="px-[10%] md:px-[15%] text-center items-center sm:pb-36 relative">
+        <section @class([ 'hidden' => count($references)==0, 'px-[10%] md:px-[15%] text-center items-center sm:pb-36 relative' ])>
```
```suggestion
        <section @class([ 'hidden' => $references->isEmpty(), 'px-[10%] md:px-[15%] text-center items-center sm:pb-36 relative' ])>
```

kamilpiech97 commented on resources/views/home.blade.php:
```diff
@@ -262,60 +260,13 @@ class="tech-icon h-full w-12 m-auto grayscale hover:grayscale-0 object-contain t
-                        linkedIn="{{ $testimonials[2]['linkedIn'] }}"
-                    />
-                </div>
+            <div @class([ 'sm:block main-gallery js-flickity' => count($references)>3, 'sm:flex justify-center' => count($references)<=3])
```
```suggestion
            <div @class([ 'sm:block main-gallery js-flickity' => $references->->count() > 3, 'sm:flex justify-center' =>  $references->->count() <= 3])
```

kamilpiech97 replied:
For me, better should be moved that into variable in controller like `$referencesCount` and use them here

kamilpiech97 commented on app/Http/Controllers/HomeController.php:
```diff
@@ -4,13 +4,26 @@
     {
-        return $factory->make("home");
+        $references = Reference::query()->where("published", true)->get();
+        $locale = App::currentLocale();
```
```suggestion
        $locale = $request->getLocale();
```
instead of Facade usage

kamilpiech97 commented on app/Http/Controllers/HomeController.php:
```diff
@@ -4,13 +4,26 @@
+        $references = Reference::query()->where("published", true)->get();
+        $locale = App::currentLocale();
+
+        $caseStudy = $locale === "en" ? json_decode(file_get_contents(public_path("main_case_study.json")), true)["en"] : json_decode(file_get_contents(public_path("main_case_study.json")), true)["pl"];
```
```suggestion
        $caseStudy = json_decode(file_get_contents(public_path("main_case_study.json")), true)[$locale];
```
hm?

kamilpiech97 commented on app/Http/Controllers/HomeController.php:
```diff
@@ -4,13 +4,26 @@
+
+        $caseStudy = $locale === "en" ? json_decode(file_get_contents(public_path("main_case_study.json")), true)["en"] : json_decode(file_get_contents(public_path("main_case_study.json")), true)["pl"];
+        $articles = json_decode(file_get_contents(public_path("articles.json")), true);
+        $referencesCount = count($references);
```
```suggestion
        $referencesCount = $references->count();
```

kamilpiech97 commented on resources/views/home.blade.php:
```diff
@@ -262,60 +255,13 @@ class="tech-icon h-full w-12 m-auto grayscale hover:grayscale-0 object-contain t
-                        linkedIn="{{ $testimonials[2]['linkedIn'] }}"
-                    />
-                </div>
+            <div @class([ 'sm:block main-gallery js-flickity' => $referencesCount>3, 'sm:flex justify-center' => $referencesCount<=3])
```
```suggestion
            <div @class([ 'sm:block main-gallery js-flickity' => $referencesCount > 3, 'sm:flex justify-center' => $referencesCount <= 3])
```

kamilpiech97 commented on app/Http/Controllers/HomeController.php:
```diff
@@ -4,13 +4,26 @@
 class HomeController extends Controller
 {
-    public function __invoke(Factory $factory): View
+    public function __invoke(Factory $factory, Request $request): View
```
```suggestion
    public function __invoke(Request $request, Factory $factory): View
```
